drivers/amazonec2/amazonec2.go

```diff
@@ -995,6 +995,9 @@ func (d *Driver) getInstance() (*ec2.Instance, error) {
 	if err != nil {
 		return nil, err
 	}
+	if len(instances.Reservations) == 0 || len(instances.Reservations[0].Instances) == 0 {
+		return nil, fmt.Errorf("instance %v not found", d.InstanceId)
+	}
```

Comment:
This and other checks below are unrelated to your PR right, just avoiding an invalid index's or gracefully handling case ?

Reply:
This is related to the scenario of deleting a machine that doesn't exist. This should not error and assume the machine is deleted.